src/rs.cpp

```diff
@@ -792,7 +792,6 @@ void rs2_get_option_range(const rs2_options* options, rs2_option option,
     float* min, float* max, float* step, float* def, rs2_error** error) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL(options);
-    VALIDATE_OPTION(options, option);
```

Comment:
I didn't understand why you removed in some places and not others.
For example both `rs2_set_option` and `rs2_get_option_range` get the option (searching in the container) and than get the option's range.

Reply:
The option has to be "supported" (read "enabled") for set_option.
Not true for get_option_range: we obviously want an option that exists, and get_option() will check that. But it doesn't have to be enabled to get its range. Unfortunately, "supported" returns false if supported but not enabled. So I removed from the metadata functions.